middle_end/flambda2.0/from_lambda/closure_conversion.ml

```diff
@@ -717,7 +717,7 @@ and close_one_function t ~external_env ~by_closure_id decl
     List.fold_left (fun (to_bind, vars_for_idents) function_decl ->
         let let_rec_ident = Function_decl.let_rec_ident function_decl in
         let to_bind, var =
-          if Ident.same our_let_rec_ident let_rec_ident then
+          if Ident.same our_let_rec_ident let_rec_ident && is_curried then
```

Comment:
Can you please add a comment about this `is_curried` case?  I'm not sure it's that obvious that the problem here relates to the tuplification wrapper.

Reply:
I've added a comment